Flowsheet_Class/flowsheet.py

```
@@ -667,66 +842,98 @@ def split_HI_nodes(self, OntoCapeNames=False):
-                                   "signal" in v.keys() if v["signal"]}
-        edges_to_remove = [k for k, v in edge_information_signal.items() if v == ["not_next_unitop"]]
-        flowsheet_wo_signals.remove_edges_from(edges_to_remove)
+                                   "signal" in v.keys() if v["signal"]} # > alright, we don't play with the keys here
+        edges_to_remove = [k for k, v in edge_information_signal.items() if v != []]  # > same
+        if self.use_single_signal_stream_old:
+            edges_to_remove = [k for k, v in edge_information_signal.items() if v != ["next_unitop"]]  # > same
+        flowsheet_wo_signals.remove_edges_from(edges_to_remove) # > alright, it is good that the MultiDiGraph keys are present here.
+
+        # First get the names of the HEX we actually have to split:
+        hex_to_split = []
+        for node_name, node_attrs in flowsheet_wo_signals.nodes(data=True):
```

Ruff (0.15.9)

[warning] Loop control variable `node_attrs` not used within loop body

(B007)